### ISSUE TITLE
Re-order Effects Blocks to Intended Order

### ISSUE THREAD

**Issue by maikito26** [dbagchee/helix-preset-viewer#15]

**Comment by maikito26** [comment]
For this I thought we could sort the preset.data.tone.dsp0/1 arrays using orderBy but apparently it doesn't work in Vue 2.x.  I couldn't get lodash's implementation to work for me.   But if we could sort by the '@position' key then we can print the blocks in order, including split/merge blocks for routing.  

https://vuejs.org/v2/guide/migration.html#Replacing-the-orderBy-Filter

Ultimately, if we could draw two path's per DSP this would help and draw the blocks per the '@path' key.

I think this will work because there is ALWAYS a SPLIT and a JOIN key at the same level of each BLOCK.  It's just a matter of placing them in the correct position by the sort.  So I would include those in the sort, and with a simple IF for each object to check as we read the array.

Then add in the INPUTs and OUTPUTs afterwards and that should take care of routing, no?

**Comment by dbagchee** [comment]
Yep - sounds correct after reviewing the JSON structure. To start I'll update the rendering so that all blocks get split into their correct sub path (@path) - I think I'll need to refactor the path template a little.

**Comment by dbagchee** [comment]
Ok split into 4 paths for now - I can refactor the css a little later to make them "sub paths". May make it easier to draw path split lines as well - now just need to figure out how to sort.

**Comment by maikito26** [comment]
Looking good so far.  Yea... that sort seems crucial.  

I did notice that Stereo pre-amp may exist on both paths in the same position?  Or at least I have a preset that seems to have a position 4 for both paths.  So might also need to think about how to display that.

**Comment by maikito26** [comment]
This one has position 4 on both paths of dsp0
[Yin Yang.txt](https://github.com/dbagchee/helix-preset-viewer/files/2508555/Yin.Yang.txt)

**Comment by dbagchee** [comment]
Cool this will be a good preset to test with:

![screen shot 2018-10-24 at 5 26 08 pm](https://user-images.githubusercontent.com/1245875/47468786-39e10980-d7b2-11e8-844e-af41247fdde8.png)

Looks like the block position count is always 8 - in this case it makes sense that the other speaker block is at position 4 so we can display it in the same spot according to it's position. For split/joins it looks like they appear just before or just after the block that is assigned to the same position number. 

Slowly but surely the logic unravels :)

**Comment by dbagchee** [comment]
After spending way too much time trying to write a working sort function yesterday - I’m thinking now that I will try to use css/flexbox ordering - since each block is aware of its own position it may work...

**Comment by maikito26** [comment]
Wow - Just looked up flexbox.   This could be the solution also!

**Comment by dbagchee** [comment]
Woah - looks like CSS worked! Will commit shortly.

**Comment by maikito26** [comment]
Wicked!  Glad you're doing all the hard stuff :)
